### PR TITLE
ProcessConfigCreator#exceptionHandlerFactory removal. Part 1:

### PR DESCRIPTION
**Pull request by nadberezny**
1. In Flink engine use ConfigurableExceptionHandler
2. In Specs use RecordingExceptionListener instead of RecordingExceptionHandler.

### CODE REVIEW

File: components/base/src/test/resources/META-INF/services/pl.touk.nussknacker.engine.flink.api.exception.FlinkEspExceptionConsumerProvider [hunk 1]
```diff
@@ -0,0 +1 @@
+pl.touk.nussknacker.engine.flink.test.RecordingExceptionConsumerProvider
```
Comment (mproch) [inline components/base/src/test/resources/META-INF/services/pl.touk.nussknacker.engine.flink.api.exception.FlinkEspExceptionConsumerProvider:1]: let's put it in flink-test-util?

File: engine/flink/kafka-util/src/test/scala/pl/touk/nussknacker/engine/kafka/source/flink/KafkaSourceFactoryProcessMixin.scala [hunk 2]
```diff
@@ -54,7 +60,7 @@ trait KafkaSourceFactoryProcessMixin extends FunSuite with Matchers with KafkaSo
   }
 
   after {
```
Comment (mproch) [inline engine/flink/kafka-util/src/test/scala/pl/touk/nussknacker/engine/kafka/source/flink/KafkaSourceFactoryProcessMixin.scala:62]: I wonder, maybe we can put it in FlinkSpec?

**Review by mproch — COMMENTED** [review]
Overall :+1: Added minor comments

File: engine/flink/engine/src/main/scala/pl/touk/nussknacker/engine/process/compiler/FlinkProcessCompiler.scala [hunk 3]
```diff
@@ -65,10 +63,8 @@ class FlinkProcessCompiler(creator: ProcessConfigCreator,
     val compiledProcess =
       ProcessCompilerData.prepare(process, definitions(processObjectDependencies), listenersToUse, userCodeClassLoader, resultCollector, runMode)
 
-    val compiledExceptionHandler = validateOrFailProcessCompilation(compiledProcess.compileExceptionHandler())
-    val listeningExceptionHandler = new ListeningExceptionHandler(listenersToUse,
-      //FIXME: remove casting...
-      compiledExceptionHandler.asInstanceOf[FlinkEspExceptionHandler])
+    val configurableExceptionHandler = new ConfigurableExceptionHandler(process.metaData, processObjectDependencies, userCodeClassLoader)
+    val listeningExceptionHandler = new ListeningExceptionHandler(listenersToUse, configurableExceptionHandler)
```
Comment (mproch) [inline engine/flink/engine/src/main/scala/pl/touk/nussknacker/engine/process/compiler/FlinkProcessCompiler.scala:67]: Let's use the runMode here to detect running in test and make `FlinkTestMain.ignore real exception handler` happy :)

Comment (mproch) [reply engine/flink/engine/src/main/scala/pl/touk/nussknacker/engine/process/compiler/FlinkProcessCompiler.scala:67]: Also: remove   `StubbedFlinkProcessCompiler.prepareExceptionHandler`

File: components/base/src/test/resources/META-INF/services/pl.touk.nussknacker.engine.flink.api.exception.FlinkEspExceptionConsumerProvider (hunk 1 above)
Comment (nadberezny) [reply components/base/src/test/resources/META-INF/services/pl.touk.nussknacker.engine.flink.api.exception.FlinkEspExceptionConsumerProvider:1]: :+1:

Comment (nadberezny) [reply components/base/src/test/resources/META-INF/services/pl.touk.nussknacker.engine.flink.api.exception.FlinkEspExceptionConsumerProvider]: https://github.com/TouK/nussknacker/pull/2484/commits/dd2818b975a391c5ddc42f2335c8be2b316042fe

File: engine/flink/kafka-util/src/test/scala/pl/touk/nussknacker/engine/kafka/source/flink/KafkaSourceFactoryProcessMixin.scala (hunk 2 above)
Comment (nadberezny) [reply engine/flink/kafka-util/src/test/scala/pl/touk/nussknacker/engine/kafka/source/flink/KafkaSourceFactoryProcessMixin.scala]: https://github.com/TouK/nussknacker/pull/2484/commits/dd2818b975a391c5ddc42f2335c8be2b316042fe

File: engine/flink/engine/src/main/scala/pl/touk/nussknacker/engine/process/compiler/FlinkProcessCompiler.scala (hunk 3 above)
Comment (nadberezny) [reply engine/flink/engine/src/main/scala/pl/touk/nussknacker/engine/process/compiler/FlinkProcessCompiler.scala]: Removed prepExcHanlder from StubbedCompiler:
https://github.com/TouK/nussknacker/pull/2484/commits/15445eea3ccc8413766d0216cde2a08876504dcb

RunMode.Test uses empty handler:
https://github.com/TouK/nussknacker/pull/2484/commits/d7911e8c0b3158e3e41a4d84c1c4223d0d87045c

**Comment by nadberezny** [comment]
@mproch thanks a lot for the review!
I've made all the changes you suggested.

**Review by mproch — APPROVED** [review]
:+1: LGTM

**Comment by nadberezny** [comment]
PR moved to:
https://github.com/TouK/nussknacker/pull/2498

because branch moved from nadberezny to touk

**Pull request closed without merging**